.github/workflows/test-nix-shell.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+cabal update hackage.haskell.org,2022-08-27T00:00:00Z
```

Comment:
Does this timestamp have to be updated manually frequently?  Should there be instructions when to do so?